### PR TITLE
Improve support for xcpretty report options

### PR DESCRIPTION
**Pull request by coreyfloyd**
This pull request addresses the following issue:
Although a user can set multiple xcpretty reports with the ":report_formats" argument, Fastlane is unable to accept multiple values for ":report_path". This means a user can only define the output path for one of the reports even if they specify more than one.

This pull request defines a new parameter ":reports", which takes an array of hashes. The keys of the has map exactly to the parameters taken by xcpretty. This enables the user to specify all options for each report they want to generate.

The old options were left intact for backwards compatibility - those can be removed if desired.

This was my best first guess at how to implement, feedback welcome.

### CODE REVIEW

File: spec/actions_specs/xcodebuild_spec.rb [hunk 1]
```diff
@@ -374,6 +374,41 @@
...
+        expect(result).to eq(
+          "set -o pipefail && " \
+          + "xcodebuild " \
+          + "-destination \"name=iPhone 5s,OS=8.1\" " \
+          + "-scheme \"MyApp\" " \
+          + "-workspace \"MyApp.xcworkspace\" " \
+          + "test " \
+          + "| xcpretty --color " \
+          + "--report html " \
+          + "--output \"./build-dir/test-report.html\" " \
```
Comment (KrauseFx) [inline spec/actions_specs/xcodebuild_spec.rb]: Drop the `\"` in this line

File: spec/actions_specs/xcodebuild_spec.rb [hunk 2]
```diff
@@ -374,6 +374,41 @@
...
+          + "--screenshots " \
+          + "--report junit " \
+          + "--output \"./build-dir/test-report.xml\" " \
```
Comment (KrauseFx) [inline spec/actions_specs/xcodebuild_spec.rb]: Same here

**Comment by KrauseFx** [comment]
Thanks for working on this. The test should be easy to fix, I added 2 comments.

**Comment by coreyfloyd** [comment]
@KrauseFx  thanks - still figuring out ruby… 

looks like it worked… but failing now for some other reason…

My local was complaining about:
```
Fastlane
  Fastlane::JUnitGenerator
    #generate
      properly generates a valid JUnit XML File (FAILED - 1)
```

But it looks like Travis is upset about something else… any ideas on how to resolve?

**Comment by KrauseFx** [comment]
That's my fault, I'll fix it, thanks!

**Comment by KrauseFx** [comment]
Thanks for working on this :+1:

**Comment by coreyfloyd** [comment]
No prob - thinks for getting it in so fast!

**Pull request merged**